lib/iris/analysis/_interpolation.py

```diff
@@ -125,7 +125,7 @@ def extend_circular_data(data, coord_dim):
     return data
 
 
-def get_xy_dim_coords(cube):
+def get_xy_dim_coords(cube, dim_coords=True):
```

Comment:
@dkillick I can see why you've done this, to get the 2d auxiliary spatial coordinates for a curvilinear grid, but now the function name is somewhat confused since it can now return non-dimensional coordinates ...